src/Core/Compiler/NodeConverter.php

```diff
@@ -192,24 +182,41 @@ protected function convertViewHelperNode(ViewHelperNode $node)
                         'execution' => $argumentValue
                     ];
                 }
-                $argumentInitializationCode .= $converted['initialization'];
+                $accumulatedArgumentInitializationCode .= $converted['initialization'];
                 $argumentInitializationCode .= sprintf(
-                    '%s[\'%s\'] = %s;',
-                    $argumentsVariableName,
+                    '\'%s\' => %s',
                     $argumentName,
                     $converted['execution']
-                ) . chr(10);
-                $alreadyBuiltArguments[$argumentName] = true;
+                );
+
+                $argumentInitializationCode .= ', ';
+            }
+
+            foreach ($node->getArgumentDefinitions() as $argumentName => $argumentDefinition) {
```

Comment:
The change of order won't cause any harm?

Reply:
No, all safe - actually there was a bad behavior before this patch: all arguments' default values would be built every time and then overwritten by subsequent assigning of the arguments. Problem was that `!isset($alreadyBuiltArguments[$argumentName])` would always be true because `$alreadyBuiltArguments` was always undefined (gets defined *after* the loop that checks it).

Now, they will only be built if not already built - just like the original behavior intended.

Reply:
Makes sense, thanks for the explanation. :-)